packages/contracts/contracts/standards/L2StandardERC721.sol

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+
+import { ERC721 } from "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import { IERC165 } from "@openzeppelin/contracts/utils/introspection/IERC165.sol";
+import { Lib_PredeployAddresses } from "../libraries/constants/Lib_PredeployAddresses.sol";
+import { Lib_Strings } from "../libraries/utils/Lib_Strings.sol";
+import "./IL2StandardERC721.sol";
+
+contract L2StandardERC721 is IL2StandardERC721, ERC721 {
+    address public l1Token;
+    address public l2Bridge;
+    string public baseTokenURI;
+
+    /**
+     * @param _l2Bridge Address of the L2 standard bridge.
+     * @param _l1Token Address of the corresponding L1 token.
+     * @param _name ERC721 name.
+     * @param _symbol ERC721 symbol.
+     */
+    constructor(
+        address _l2Bridge,
+        address _l1Token,
+        string memory _name,
+        string memory _symbol
+    ) ERC721(_name, _symbol) {
+        l1Token = _l1Token;
+        l2Bridge = _l2Bridge;
+
+        // Creates a base URI in the format specified by EIP-681:
+        // https://eips.ethereum.org/EIPS/eip-681
+        baseTokenURI = string(
+            abi.encodePacked(
+                "ethereum:0x",
+                Lib_Strings.addressToString(_l1Token),
+                "@1/tokenURI?uint256="
```

Comment:
`block.chainid` should be used instead of hardcoding `1` here so that this can work with testnets and L3s